packages/network-controller/src/NetworkController.ts

```diff
@@ -917,22 +917,29 @@ export class NetworkController extends BaseControllerV2<
 
     const isEIP1559Compatible = await this.#determineEIP1559Compatibility();
     this.update((state) => {
-      state.networkDetails.EIPS[1559] = isEIP1559Compatible;
+      if (isEIP1559Compatible !== undefined) {
```

Comment:
Nit: Perhaps we could skip the state update altogether in this case? I'm not sure but I believe this would still trigger a state update event, despite the state not changing.